src/beast/beast/http/rfc2616.h

```diff
@@ -253,6 +282,183 @@ split_commas(std::string const& s)
     return split_commas(s.begin(), s.end());
 }
 
+//------------------------------------------------------------------------------
+
+/** Iterates through a comma separated list.
+
+    Meets the requirements of ForwardIterator.
+
+    List defined in rfc2616 2.1.
+
+    @note Values returned may contain backslash escapes.
+*/
+class list_iterator
+{
+    using iter_type = string_view::const_iterator;
+
+    iter_type it_;
+    iter_type end_;
+    string_view value_;
+
+public:
+    using value_type = string_view;
+    using pointer = value_type const*;
+    using reference = value_type const&;
+    using difference_type = std::ptrdiff_t;
+    using iterator_category =
+        std::forward_iterator_tag;
+
+    list_iterator(iter_type begin, iter_type end)
+        : it_(begin)
+        , end_(end)
+    {
+        if(it_ != end_)
+            increment();
+    }
+
+    bool
+    operator==(list_iterator const& other) const
+    {
+        return other.it_ == it_ && other.end_ == end_
+            && other.value_.size() == value_.size();
+    }
+
+    bool
+    operator!=(list_iterator const& other) const
+    {
+        return !(*this == other);
+    }
+
+    reference
+    operator*() const
+    {
+        return value_;
+    }
+
+    pointer
+    operator->() const
+    {
+        return &*(*this);
+    }
+
+    list_iterator&
+    operator++()
+    {
+        increment();
+        return *this;
+    }
+
+    list_iterator
+    operator++(int)
+    {
+        auto temp = *this;
+        ++(*this);
+        return temp;
+    }
+
+private:
+    void
+    increment()
+    {
+        value_.clear();
+        while(it_ != end_)
+        {
+            if(*it_ == '"')
+            {
+                // quoted-string
+                ++it_;
+                if(it_ == end_)
+                    return;
+                if(*it_ != '"')
+                {
+                    auto start = it_;
+                    for(;;)
+                    {
+                        ++it_;
+                        if(it_ == end_)
+                        {
+                            value_ = string_view(
+                                &*start, std::distance(start, it_));
+                            return;
+                        }
+                        if(*it_ == '"')
+                        {
+                            value_ = string_view(
+                                &*start, std::distance(start, it_));
+                            ++it_;
+                            return;
+                        }
+                    }
+                }
+                ++it_;
+            }
+            else if(*it_ == ',')
```

Comment:
Maybe define `is_sep` that returns true if the char is a separator - either lws or comma?